pandas/tests/indexes/common.py

```diff
@@ -933,3 +933,10 @@ def test_empty(self):
         index = self.create_index()
         assert not index.empty
         assert index[:0].empty
+
+    @pytest.mark.parametrize('how', ['outer', 'inner', 'left', 'right'])
```

Comment:
can you also add the tests from the original issue (it was on PI) as a specific test.